src/hooks/use-debounced-block.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { ConnectedProvider } from "src/domain";
+import { AsyncTask } from "src/utils/types";
+
+type DebouncedBlockCallback = (blockNumber: number) => void;
+
+const DEBOUNCE_TIME_IN_MS = 750;
+
+const useDebouncedBlock = (connectedProvider: AsyncTask<ConnectedProvider, string>) => {
+  const [listener, setListener] = useState<DebouncedBlockCallback>();
+  const [blockNumber, setBlockNumber] = useState<number>();
+
+  const addBlockListener = useCallback(
+    (cb: DebouncedBlockCallback) => {
+      if (!listener) {
+        setListener(() => cb);
```

Comment:
It's not a smell, it's how `setState` works. If you pass a callback to the `setState` it will work as `setState((oldValue) => // ...)`, so it's not setting anything if you pass it a callback.

Reply:
I don't understand what you mean.

Anyway, this is gone now.